Sources/client/Socket.swift

```diff
@@ -147,51 +150,36 @@ public class Socket {
   //----------------------------------------------------------------------
   // MARK: - Initialization
   //----------------------------------------------------------------------
+  public convenience init(_ endPoint: String) {
+    self.init(endPoint: endPoint,
+              transport: { url in return WebSocket(url: url) },
+              params: { nil })
+  }
+
+  @available(*, deprecated, message: "Deprecated in favour of Socket(_: String, paramsClosure: PayloadClosure?) instead")
+  public convenience init(_ endPoint: String,
+                          params: Payload? = nil) {
+    self.init(endPoint: endPoint,
+              transport: { url in return WebSocket(url: url) },
+              params: { params })
+  }
+
   public convenience init(_ endPoint: String,
-                          params: [String: Any]? = nil) {
+                          paramsClosure: PayloadClosure? = nil) {
```

Comment:
I wouldn't make this optional. If the client wants to provide a params closure, then they have to specifically declare they are doing so.